ui/imodel-components-react/src/imodel-components-react/editors/NewColorEditor.tsx

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import {
+  type ColorEditorParams,
+  StandardEditorNames,
+} from "@itwin/appui-abstract";
+import {
+  ColorPalette,
+  ColorPicker,
+  ColorSwatch,
+  ColorValue,
+  IconButton,
+  Popover,
+} from "@itwin/itwinui-react";
+import type {
+  EditorProps,
+  EditorSpec,
+  NumericValue,
+  RequiredProps,
+  ValueMetadata,
+} from "@itwin/components-react";
+import { isNumericValue } from "@itwin/components-react";
+
+/**
+ * Editor specification for color editor.
+ * @beta
+ */
+export const ColorEditorSpec: EditorSpec = {
+  applies: (metadata) =>
+    metadata.type === "number" &&
+    metadata.preferredEditor === StandardEditorNames.ColorPicker &&
+    (metadata as ColorValueMetadata).params !== undefined,
+  Editor: ColorEditor,
+};
+
+/**
+ * Metadata for color editor.
+ * @beta
+ */
+export interface ColorValueMetadata extends ValueMetadata {
+  params: ColorEditorParams[];
```

Comment:
why is this an array?

Reply:
Switched to `params: PropertyEditorParams[];`. This is what is passed to old editors through `PropertyRecord`